contrib/monitored_resource_util/src/main/java/io/opencensus/contrib/monitoredresource/util/MonitoredResource.java

```diff
@@ -102,9 +102,29 @@ public ResourceType getResourceType() {
      */
     public abstract String getRegion();
 
-    static AwsEc2InstanceMonitoredResource create() {
+    /**
+     * Returns an {@link AwsEc2InstanceMonitoredResource}.
+     *
+     * @param account the AWS account ID.
+     * @param instanceId the AWS EC2 instance ID.
+     * @param region the AWS region.
+     * @return an {@code AwsEc2InstanceMonitoredResource}.
+     * @since 0.15
+     */
+    public static AwsEc2InstanceMonitoredResource create(
+        String account, String instanceId, String region) {
       return new AutoValue_MonitoredResource_AwsEc2InstanceMonitoredResource(
-          AWS_ACCOUNT, AWS_INSTANCE_ID, AWS_REGION);
+          account, instanceId, region);
+    }
+
+    /**
+     * Returns an {@link AwsEc2InstanceMonitoredResource} with the default parameters.
+     *
+     * @return an {@code AwsEc2InstanceMonitoredResource}.
+     * @since 0.15
+     */
+    public static AwsEc2InstanceMonitoredResource create() {
```

Comment:
Could we continue hiding the no-arg factory methods, to reduce the size of the API?  I also think that the no-arg methods could be confusing, because users might expect the factory methods in AutoValue classes to be deterministic, but these depend on the environment.

Reply:
Make sense to me, done.